cmd/regex_format.go

```diff
@@ -164,13 +167,39 @@ func processFile(filePath string, ctxt *processors.Context, checkOnly bool) erro
 		return err
 	}
 
-	parser := parser.NewParser(ctxt, file)
-	parsedBytes, _ := parser.Parse(true)
+	raParser := parser.NewParser(ctxt, file)
+	parsedBytes, _ := raParser.Parse(true)
 	if err = file.Close(); err != nil {
 		logger.Error().Err(err).Msgf("file already closed %s", filePath)
 		return err
 	}
 
+	// sanity check: if we are using an ignore-case flag, we don't need to have any uppercase letters in the file
+	if raParser.Flags['i'] {
+		found := bytes.IndexAny(parsedBytes.Bytes(), "ABCDEFGHIJKLMNOPQRSTUVWXYZ")
+		if found > -1 {
+			// print bytes and show where the violation is
+			showLast := found + showCharsAround
+			showFirst := found - showCharsAround
+			lastByte := len(parsedBytes.Bytes())
+			if showLast > lastByte {
+				showLast = lastByte
+			}
+			if showFirst < 0 {
+				showFirst = 0
+			}
+			logger.Warn().Msgf("File contains uppercase letters, but ignore-case flag is set. Please check your source files.")
+			logger.Warn().Msgf("Problem found around char %d:\n%s\n", found, parsedBytes.String()[showFirst:showLast])
+			logger.Warn().Msg("Be aware that because of file inclusions and definitions, the actual line number or file might be different.")
+			message := "File not properly formatted"
```

Comment:
I would only print the message and return the value if `checkOnly` is true to be consistent with the behaviour below.

Reply:
Sure. Did reorganize things a bit to DRY.